internal/static/cartogram2.js

```diff
@@ -1026,8 +1166,574 @@ class CartMap {
+                .attr("dy", "-10px");
+            legendText.append('tspan')
+                .text(unit)
+                .attr("dy", "10px")
+                .attr("dx", "8px");
+        }
+
+        // Event for when a different legend size is selected.
+        const legendNumber = d3.select("#" + legendSVGID + "-number").text();
+
+        const changeToC = () => {
+
+            // Update currentGridPath in other versions
+            if(legendSVGID != "map-area-legend") {
+                Object.keys(this.versions).forEach(function (version_sysname) {
+                        this.versions[version_sysname].legendData["currentGridPath"] = "gridC";
+                }, this);
+            } else {
+                this.versions[sysname].legendData["mapCurrentGridPath"] = "gridC"
+            }
+
+            d3.select("#" + legendSVGID + "C").attr('fill', '#FFFFFF');
+            d3.select("#" + legendSVGID + "B").attr('fill', '#FFFFFF');
+            d3.select("#" + legendSVGID + "A").attr('fill', '#FFFFFF');
+
+            d3.select("#" + legendSVGID.substring(0, legendSVGID.length-6) + "grid")
+                .transition()
+                .duration(1000)
+                .attr('d', gridC);
+
+            d3.select("#" + legendSVGID + "-number")
+              .text(parseInt(legendNumber.substring(0,1))/scaleNiceNumberA*scaleNiceNumberC + legendNumber.substring(1, legendNumber.length));
+        }
+
+        const changeToB = () => {
+            // Update currentGridPath in other versions
+            if(legendSVGID != "map-area-legend") {
+                Object.keys(this.versions).forEach(function (version_sysname) {
+                        this.versions[version_sysname].legendData["currentGridPath"] = "gridB";
+                }, this);
+            } else {
+                this.versions[sysname].legendData["mapCurrentGridPath"] = "gridB"
+            }
+
+            d3.select("#" + legendSVGID + "C").attr('fill', '#EEEEEE');
+            d3.select("#" + legendSVGID + "B").attr('fill', '#FFFFFF');
+            d3.select("#" + legendSVGID + "A").attr('fill', '#FFFFFF');
+
+            d3.select("#" + legendSVGID.substring(0, legendSVGID.length-6) + "grid")
+              .transition()
+              .duration(1000)
+              .attr('d', gridB);
+
+            d3.select("#" + legendSVGID + "-number")
+              .text(parseInt(legendNumber.substring(0,1))/scaleNiceNumberA*scaleNiceNumberB + legendNumber.substring(1, legendNumber.length));
+        }
+
+        const changeToA = () => {
+            // Update currentGridPath in other versions
+            if(legendSVGID != "map-area-legend") {
+                Object.keys(this.versions).forEach(function (version_sysname) {
+                        this.versions[version_sysname].legendData["currentGridPath"] = "gridA";
+                }, this);
+            } else {
+                this.versions[sysname].legendData["mapCurrentGridPath"] = "gridA"
+            }
```

Comment:
Since I suggest you don't store `currentGridPath` and `mapCurrentGridPath` in the `MapVersion` instance, here are two alternatives:

1. When the user changes the legend, assign a new event handler to `gridline-toggle-MAPNAME` that will toggle the appropriate grid version (A, B, or C) when enabled, or

2. Set an attribute on the legend SVG element with the current grid version:
    ```javascript
    document.getElementById(legendSVGID).dataset.currentGridPath = "gridA";
    ```
    Then, in your `drawGridLines` function, you can make reference to  `document.getElementById(legendSVGID).dataset.currentGridPath` to get the correct grid path to draw. For more information on how HTML element datasets work, you can see the [MDN documentation page](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) for them.